### PR TITLE
Finish switch to self-hosted RPM repositories.

### PR DESCRIPTION
**Pull request by Ferroin**
##### Summary

Followup to #14100, this updates the kickstart script to use the new infrastructure, and adds documentation about how to manually set up our RPM repositories.

##### Test Plan

Running the kickstart script from this PR on a platform we provide native for with the `--native-only` switch should correctly install the agent.

##### Additional Information

The new documentation added by this PR will be further expanded when we have working repositories for DEB packages on the new infrastructure.

### CODE REVIEW

File: packaging/installer/kickstart.sh [hunk 1]
```diff
@@ -28,8 +28,10 @@ KICKSTART_SOURCE="$(
 PACKAGES_SCRIPT="https://raw.githubusercontent.com/netdata/netdata/master/packaging/installer/install-required-packages.sh"
 PATH="${PATH}:/usr/local/bin:/usr/local/sbin"
 PUBLIC_CLOUD_URL="https://app.netdata.cloud"
-REPOCONFIG_URL_PREFIX="https://packagecloud.io/netdata/netdata-repoconfig/packages"
-REPOCONFIG_VERSION="1-2"
+REPOCONFIG_DEB_URL_PREFIX="https://packagecloud.io/netdata/netdata-repoconfig/packages"
+REPOCONFIG_DEB_VERSION="1-2"
+REPOCONFIG_RPM_URL_PREFIX="https://repo.netdata.cloud/repos/repoconfig/"
+REPOCONFIG_RPM_VERSION="2-1"
```
Comment (thiagoftsm) [inline packaging/installer/kickstart.sh:34]: @Ferroin the `shellcheck` warnings do not have relationship with this PR, but if it is possible, please, make another PR removing them.

File: packaging/installer/kickstart.sh [hunk 2]
```diff
@@ -28,8 +28,10 @@ KICKSTART_SOURCE="$(
 PACKAGES_SCRIPT="https://raw.githubusercontent.com/netdata/netdata/master/packaging/installer/install-required-packages.sh"
 PATH="${PATH}:/usr/local/bin:/usr/local/sbin"
 PUBLIC_CLOUD_URL="https://app.netdata.cloud"
-REPOCONFIG_URL_PREFIX="https://packagecloud.io/netdata/netdata-repoconfig/packages"
-REPOCONFIG_VERSION="1-2"
+REPOCONFIG_DEB_URL_PREFIX="https://packagecloud.io/netdata/netdata-repoconfig/packages"
+REPOCONFIG_DEB_VERSION="1-2"
+REPOCONFIG_RPM_URL_PREFIX="https://repo.netdata.cloud/repos/repoconfig/"
```
Comment (Dim-P) [inline packaging/installer/kickstart.sh:33]: Should remove the trailing slash here.

File: packaging/installer/kickstart.sh [hunk 3]
```diff
@@ -1468,8 +1470,17 @@ try_package_install() {
...
-  repoconfig_url="${REPOCONFIG_URL_PREFIX}/${repo_prefix}/${repoconfig_file}/download.${pkg_type}"
+
+  case "${pkg_type}" in
+    deb)
+      repoconfig_file="${repoconfig_name}${pkg_vsep}${REPOCONFIG_DEB_VERSION}${pkg_suffix}.${pkg_type}"
+      repoconfig_url="${REPOCONFIG_DEB_URL_PREFIX}/${repo_prefix}/${repoconfig_file}/download.${pkg_type}"
+      ;;
+    rpm)
+      repoconfig_file="${repoconfig_name}${pkg_vsep}${REPOCONFIG_RPM_VERSION}${pkg_suffix}.${pkg_type}"
+      repoconfig_url="${REPOCONFIG_RPM_URL_PREFIX}/${repo_prefix}/${repoconfig_file}/${repoconfig_file}"
```
Comment (Dim-P) [inline packaging/installer/kickstart.sh:1481]: I think `repoconfig_url` is wrong, please check the log when I was testing on Fedora 37:
```
 --- Checking for availability of repository configuration package. ---
+ check_for_remote_file https://repo.netdata.cloud/repos/repoconfig//fedora/37/netdata-repo-edge-2-1.no
arch.rpm/netdata-repo-edge-2-1.noarch.rpm
+ url=https://repo.netdata.cloud/repos/repoconfig//fedora/37/netdata-repo-edge-2-1.noarch.rpm/netdata-r
epo-edge-2-1.noarch.rpm
```

**Review by Dim-P — CHANGES_REQUESTED** [review]
Some issues with the new URL while testing on Fedora 37. On Ubuntu 22.04 it still works as it should, using the packagecloud repos.

Nice `packages.md` !

File: packaging/installer/kickstart.sh (hunk 3 above)
Comment (ilyam8) [reply packaging/installer/kickstart.sh:1481]: The _problem_ is `REPOCONFIG_RPM_URL_PREFIX`,  it has trailing `/` (`REPOCONFIG_DEB_URL_PREFIX` has no).

Comment (ilyam8) [inline packaging/installer/kickstart.sh:1481]:
```suggestion
      repoconfig_url="${REPOCONFIG_RPM_URL_PREFIX}/${repo_prefix}/${SYSARCH}/${repoconfig_file}"
```

Fixes

```
 WARNING  No repository configuration package available for fedora 37. Cannot install native packages on this system
```

**Review by ilyam8 — CHANGES_REQUESTED** [review]

File: packaging/installer/kickstart.sh (hunk 2 above)
Comment (Ferroin) [reply packaging/installer/kickstart.sh:33]:
```suggestion
REPOCONFIG_RPM_URL_PREFIX="https://repo.netdata.cloud/repos/repoconfig"
```

File: packaging/installer/methods/packages.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+Within each top level group of repositories, there are directories for each supported group of distributions:
+
+- `el`: Is for Red Hat Enterprise Linux and binary compatible distros, such as CentOS, Alma Linux, and Rocky Linux.
+- `fedora`: Is for Fedora and binary compatible distros.
+- `ol`: Is for Oracle Linux and binary compatible distros.
+- `opensuse`: Is for openSUSE and binary compatible distros.
+
+Under each of those directories is a directory for each supported release of that distribution, and under that a
+directory for each supported CPU architecture which contains tthe actual repository.
```
[the remark this answers is not in the document]
Comment (Ferroin) [reply packaging/installer/methods/packages.md:36]:
```suggestion
directory for each supported CPU architecture which contains the actual repository.
```

File: packaging/installer/kickstart.sh [hunk 5]
```diff
@@ -1468,8 +1470,17 @@ try_package_install() {
...
+  case "${pkg_type}" in
+    deb)
+      repoconfig_file="${repoconfig_name}${pkg_vsep}${REPOCONFIG_DEB_VERSION}${pkg_suffix}.${pkg_type}"
+      repoconfig_url="${REPOCONFIG_DEB_URL_PREFIX}/${repo_prefix}/${repoconfig_file}/download.${pkg_type}"
+      ;;
+    rpm)
+      repoconfig_file="${repoconfig_name}${pkg_vsep}${REPOCONFIG_RPM_VERSION}${pkg_suffix}.${pkg_type}"
+      repoconfig_url="${REPOCONFIG_RPM_URL_PREFIX}/${repo_prefix}/${SYSARCH}/${repoconfig_file}"
+      ;;
+  esac
```
Comment (ilyam8) [inline packaging/installer/kickstart.sh:1483]: I think we need to add `*` case + fatal (having anything except `deb` and `rpm` unexpected but possible due to a bug).

Comment (Ferroin) [reply packaging/installer/kickstart.sh:1483]: From a completeness perspective I agree, but in practice this will only make the messaging in such a failure a bit friendlier to the user (we will still hit a reasonably ‘safe’ failure case very quickly without a default case).

Given the urgency of getting this merged, I would rather add this in a follow-up PR as opposed to requiring yet another round of testing.

Comment (ilyam8) [reply packaging/installer/kickstart.sh:1483]: Mmm, (not handled) `*` case now:

 - `repoconfig_url` not set
 - `check_for_remote_file` fails
 - => _Could not install native binary packages, falling back to alternative installation method_

I am not sure that falling back to an alternative installation method is the desired result in this case.

**Review by ilyam8 — APPROVED** [review]

File: packaging/installer/kickstart.sh (hunk 5 above)
Comment (ilyam8) [reply packaging/installer/kickstart.sh:1483]: That is not technically possible with the current code (we always set `pkg_type` to either `deb` or `rpm`, but a typo or mistake in the future can be a problem).

**Review by MrZammler — APPROVED** [review]

**Review by thiagoftsm — APPROVED** [review]
Tested on CentOS 7 and ALma 8, LGTM!
